### DISCUSSION TITLE
Is it possible to get ProjectID of Dapps

### DISCUSSION THREAD

**Discussion by jaripy** [WalletConnect/walletconnect-monorepo#3510] [Developer Support] [answered]
We are currently working on a Wallet application.

When connecting to Dapps, we would like to get the ProjectID of the Dapps to determine if they are reliable Dapps.
I believe the Dapps side instances the `ProjectID` and `RelayURL` [here](https://docs.walletconnect.com/2.0/api/sign/dapp-usage#initialization-1), but is it possible to define and receive the `ProjectID` and `RelayURL` [here](https://docs.walletconnect.com/2.0/api/sign/wallet-usage#setting-up-event-listeners) on the Wallet side?

**Comment by glitch-txs** [comment]
hey, we're doing this under the hood, I think you'd like to check [Verify API](https://docs.walletconnect.com/2.0/web3wallet/verify) :)

**Reply by jaripy** [reply]
Thank you for your reply.

Can the [Verify API](https://docs.walletconnect.com/2.0/web3wallet/verify) be used in combination with SignClient APIs?
We would like to use `session client` and `session proposal` of [SignClient APIs](https://docs.walletconnect.com/2.0/api/sign/wallet-usage), but even if I use them, is it possible to combine them with Web3Wallet's verify API?

The example [here](https://docs.walletconnect.com/2.0/web3wallet/verify#implementation) is only for Web3Wallet's `auth_request`, so I would like to check if it is possible to use session-bases APIs as well.

Also, is it correct to assume that the verified `Domain` can be uniquely determined as shown [here](https://docs.walletconnect.com/2.0/cloud/verify)? Is it possible that there are suspicious Dapps that are disguised as the same `Domain`?

**Reply by jakubuid** [reply]
Yes, Verify API works on these threes cases:
- auth-requests
- session-proposals
- session-requests

Web3Wallet is enough to handle these by getting the VerifyContext

**Reply by jakubuid** [reply]
Web3Wallet is a combination of Auth API and Sign API, so both of them are integrated with Verify

**Reply by jaripy** [reply]
Thank you.

Am I correct in assuming that the `Domain` verified by WalletConnect is **uniquely** tied to Dapps or Wallet?

**Reply by crypblizz8** [reply, marked as answer]
Dapp side
